lib/v8.js

```diff
@@ -247,29 +245,40 @@ Deserializer.prototype.readRawBytes = function readRawBytes(length) {
                         length);
 };
 
-/* Keep track of how to handle different ArrayBufferViews.
- * The default Serializer for Node does not use the V8 methods for serializing
- * those objects because Node's `Buffer` objects use pooled allocation in many
- * cases, and their underlying `ArrayBuffer`s would show up in the
- * serialization. Because a) those may contain sensitive data and the user
- * may not be aware of that and b) they are often much larger than the `Buffer`
- * itself, custom serialization is applied. */
-const arrayBufferViewTypes = [Int8Array, Uint8Array, Uint8ClampedArray,
-                              Int16Array, Uint16Array, Int32Array, Uint32Array,
-                              Float32Array, Float64Array, DataView];
-
-const arrayBufferViewTypeToIndex = new SafeMap();
-
-{
-  const dummy = new ArrayBuffer();
-  ArrayPrototypeForEach(arrayBufferViewTypes, (ctor, i) => {
-    const tag = ObjectPrototypeToString(new ctor(dummy));
-    arrayBufferViewTypeToIndex.set(tag, i);
-  });
+function arrayBufferViewTypeToIndex(abView) {
+  const type = ObjectPrototypeToString(abView);
```

Comment:
Is `abView` coming from the user? In that case, we might want to switch to `TypedArrayPrototypeGetSymbolToStringTag()` instead.